tidy3d/components/simulation.py

```diff
@@ -2622,7 +2627,7 @@ def custom_datasets(self) -> List[Dataset]:
         datasets_source = [
             src.field_dataset for src in self.sources if isinstance(src, CustomFieldSource)
         ]
-        datasets_medium = [mat.eps_dataset for mat in self.mediums if isinstance(mat, CustomMedium)]
+        datasets_medium = [mat for mat in self.mediums if isinstance(mat, AbstractCustomMedium)]
```

Comment:
Now that custom medium no longer has a unified dataset, I'm not sure what to include here. This modification is to make sure the webapi works.

Reply:
Maybe this is the source of the from_file code failing? Could you check if the datasets_medium are what you'd expect here? Maybe the datasets are not getting written to the file in the first place?

Reply:
Seems like this custom_datasets property does two things:
1. triggers the upload of hdf5 files in the simulation task (if any datasets present)
2. gives something that can be easily validated to ensure that datasets are not None after loading.

Reply:
Does something like
```py
datasets_medium = [mat.eps_dataset for mat in self.mediums if isinstance(mat, AbstractCustomMedium)]
```
work?

Reply:
`AbstractCustomMedium` itself seems to fulfill the two things, so it passes the tests. Regarding `mat.eps_dataset`, the issue is that only `CustomMedium` contains `eps_dataset`.